### PR TITLE
[OKU-23]-Apply linters and husky in FE

### PR DESCRIPTION
**Pull request by twentymurial33**
## Change Type
*Put an `x` in all the boxes that apply
- [] Bugfix (non-breaking change which fixes an issue)
- [X] Feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to change)

## Linear issue link
https://linear.app/oku-no-kumo/issue/OKU-23/apply-linters-and-husky-in-fe

### CODE REVIEW

File: project_app/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        "prettier",
+        "next/core-web-vitals",
+        "plugin:@typescript-eslint/recommended",
+        "airbnb",
+        "prettier",
+        "plugin:jsx-a11y/recommended",
+        "plugin:prettier/recommended",
+        "plugin:sonarjs/recommended",
+        "plugin:security/recommended",
+        "plugin:react-hooks/recommended",
```
Comment (idiglove) [inline project_app/.eslintrc.js:41]: this is a lot of plugins 🤔 what do you think? @enricodelarosa

File: project_app/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        ],
+        "react/jsx-props-no-spreading": "off",
+        "import/extensions": [
+            "error",
+            "ignorePackages",
+            {
+                js: "never",
+                jsx: "never",
+                ts: "never",
+                tsx: "never",
```
Comment (idiglove) [inline project_app/.eslintrc.js:62]: what are these never for?

File: project_app/package.json [hunk 3]
```diff
@@ -6,7 +6,11 @@
     "dev": "next dev",
     "build": "next build",
     "start": "next start",
-    "lint": "next lint",
+    "check-types": "tsc --pretty --noEmit",
+    "check-format": "prettier --check .",
+    "check-lint": "eslint . --ext ts --ext tsx --ext js",
+    "format": "prettier --write .",
+    "test-all": "npm run check-format && npm run check-lint && npm run check-types && npm run build",
```
Comment (idiglove) [inline project_app/package.json:13]: does it work in a way when you commit it lints and fixes it?

File: project_app/.eslintrc.js (hunk 1 above)
Comment (twentymurial33) [reply project_app/.eslintrc.js:41]: I have some intsalled twice - I will remove the duplication

File: project_app/.eslintrc.js (hunk 2 above)
Comment (twentymurial33) [reply project_app/.eslintrc.js:62]: removed them

File: project_app/package.json (hunk 3 above)
Comment (twentymurial33) [reply project_app/package.json:13]: Yes - re-run the application today and I see format changes

**Review by idiglove — APPROVED** [review]
Good. Just be sure it runs when you git commit

**Pull request merged**